src/logging.c

```diff
@@ -99,49 +148,96 @@ rcutils_ret_t rcutils_logging_initialize_with_allocator(rcutils_allocator_t allo
     g_rcutils_logging_output_handler = &rcutils_logging_console_output_handler;
     g_rcutils_logging_default_logger_level = RCUTILS_DEFAULT_LOGGER_DEFAULT_LEVEL;
 
-    // Check the environment variable for line buffered output
-    const char * line_buffered;
+    const char * line_buffered = NULL;
     const char * ret_str = rcutils_get_env("RCUTILS_CONSOLE_STDOUT_LINE_BUFFERED", &line_buffered);
-
     if (NULL == ret_str) {
-      if (strcmp(line_buffered, "1") == 0) {
-        g_force_stdout_line_buffered = true;
-      } else if (strcmp(line_buffered, "0") != 0 && strcmp(line_buffered, "") != 0) {
+      if (strcmp(line_buffered, "") != 0) {
         fprintf(
           stderr,
-          "Warning: unexpected value [%s] specified for RCUTILS_CONSOLE_STDOUT_LINE_BUFFERED. "
-          "Default value 0 will be used. Valid values are 1 or 0.\n",
-          line_buffered);
+          "RCUTILS_CONSOLE_STDOUT_LINE_BUFFERED is now ignored.  "
+          "Please set RCUTILS_CONSOLE_USE_STDOUT and RCUTILS_CONSOLE_BUFFERED_STREAM "
+          "to control the stream and the buffering of log messages.\n");
       }
     } else {
       fprintf(
-        stderr, "Error getting env. variable "
+        stderr, "Error getting environment variable "
         "RCUTILS_CONSOLE_STDOUT_LINE_BUFFERED: %s\n", ret_str);
-      ret = RCUTILS_RET_INVALID_ARGUMENT;
+      RCUTILS_SET_ERROR_MSG_WITH_FORMAT_STRING(
+        "Error getting environment variable RCUTILS_CONSOLE_STDOUT_LINE_BUFFERED: %s", ret_str);
+      return RCUTILS_RET_ERROR;
+    }
+
+    // Set the default output stream for all severities to stderr so that errors
+    // are propagated immediately.  The user can choose to set the output stream
+    // to stdout by setting the RCUTILS_CONSOLE_USE_STDOUT environment
+    // variable to 1.
+    enum rcutils_get_env_retval retval = rcutils_get_env_var_zero_or_one(
+      "RCUTILS_CONSOLE_USE_STDOUT", "use stderr",
+      "use stdout");
```

Comment:
Nitpick: why wrap this line?

Reply:
That's what either cpplint or uncrustify wanted.

Reply:
I don't see why that would be the case. The line containing all arguments (line 175+176) contains way less than 80 characters.

Reply:
I've done...something else here that seems to make uncrustify happy.